### PR TITLE
inspircd: fix build on linux, enable more extra modules, and fix configure flags

### PR DESCRIPTION
**Pull request by sadiepowell**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

I'm the upstream for this software. Since package options were removed in 2018 we've encountered several users who have tried to use this package but have hit the issue of it not shipping any TLS support, SQL modules, and no modern regex modules. I've fixed this and also fixed the build under Linuxbrew and synced the configure options with what we recommend that packagers use.

### CODE REVIEW

File: Formula/inspircd.rb [hunk 1]
```diff
@@ -18,13 +18,22 @@ class Inspircd < Formula
...
   skip_clean "data"
   skip_clean "logs"
 
   def install
-    system "./configure", "--enable-extras=ldap"
-    system "./configure", "--prefix=#{prefix}"
+    system "./configure", "--enable-extras",
+                          "argon2 ldap mysql pgsql regex_posix regex_stdlib ssl_gnutls sslrehashsignal"
+    system "./configure", "--disable-auto-extras",
+                          "--distribution-label", "homebrew-#{revision}",
```
Comment (carlocab) [inline Formula/inspircd.rb:35]: `revision` here currently returns `0`. Is that your intention with this?

**Review by carlocab — COMMENTED** [review]
Thanks for your PR, @SadieCat. Can you add `revision 1` right after `license`, so that users get the updated version of this formula when they do `brew upgrade`?

File: Formula/inspircd.rb (hunk 1 above)
Comment (SadieCat) [reply Formula/inspircd.rb:36]: Yes. It's for uniquely identifying the revision of the package with which the software was built.

**Comment by SadieCat** [comment]
> Thanks for your PR, @SadieCat. Can you add `revision 1` right after `license`, so that users get the updated version of this formula when they do `brew upgrade`?

Done.

**Review by carlocab — APPROVED** [review]
Thanks, @SadieCat.

**Pull request closed without merging**